cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -583,17 +646,22 @@ private static DatasetId getMetadataDatasetInstance(MetadataScope scope) {
     return MetadataScope.USER == scope ? BUSINESS_METADATA_INSTANCE_ID : SYSTEM_METADATA_INSTANCE_ID;
   }
 
+
   public static MetadataDataset getMetadataDataset(DatasetContext context, DatasetFramework dsFramework,
                                                    MetadataScope scope) {
     try {
-      return DatasetsUtil
-        .getOrCreateDataset(context, dsFramework, getMetadataDatasetInstance(scope), MetadataDataset.class.getName(),
-                            DatasetProperties.builder().add(MetadataDatasetDefinition.SCOPE_KEY, scope.name()).build());
+      return DatasetsUtil.getOrCreateDataset(context, dsFramework, getV2MetadataDatasetInstance(scope),
```

Comment:
there is a slight chance that you recreate the V!1 dataset after it has just been deleted by the migration thread. Better to use getDataset() only for V1